Changes to taform for MVP

amyleblond opened the pull request.

amyleblond committed:
Changes to taform for MVP

amyleblond committed:
adding dynamic links to course information

j2kan commented on taworks/taform/static/taform/css/style.css:
```diff
@@ -79,6 +79,10 @@ table {
 .tg .tg-yw4l {
     vertical-align: top
     }
+.tg .test{
```
remove if we aren't using it. We should try to make better use of better naming for future styling

j2kan commented on taworks/taform/templates/taform/application.html:
```diff
@@ -7,6 +7,39 @@
+
+<ul>
+    <li>the course instructor rankings of applicants</li>
+    <li>the teaching support needs within the department and actual enrolment numbers</li>
```
enrollment -american spelling

lauradobson replied:
I think this was a direct copy from Smucker's current form @j2kan

amyleblond replied:
it was ^

j2kan commented on taworks/taform/templates/taform/application.html:
```diff
@@ -7,6 +7,39 @@
+    <li>Teaching assistants must be currently enrolled as students.</li>
+    <li>Teaching assistants must be fully available for their assigned course.
+        <ul>
+            <li>At a minimum, the TA must be available during times scheduled for the course's tutorial and labs.   Some instructors may need a TA available during lectures for in class exams, etc.</li>
```
extra space between
labs.   Some

j2kan commented on taworks/taform/templates/taform/application.html:
```diff
@@ -7,6 +7,39 @@
+    </li>
+    <li>Students should not have any outstanding English Language Proficiency requirements as part of their conditions of admission for their Graduate Studies program.</li>
+    <li>A Full-time registered graduate student is not permitted to be employed by the University for more than an average of (10) ten hours per week per term.</li>
+    <li>nternational students must have a valid study permit.</li>
```
Missing "I" in international

j2kan commented on taworks/taform/templates/taform/application.html:
```diff
@@ -79,10 +112,10 @@
         <li> <b>1:</b> I prefer to teach this course</li>
         <li> <b>2:</b> I am able to teach this course</li>
         <li> <b>3:</b> I would prefer not to teach this course</li>
+        <li><b>0:</b> I am unable to TA this course</li>
```
add space after <li> to be consistent

j2kan commented on taworks/taform/templates/taform/application.html:
```diff
@@ -96,7 +129,8 @@
     <div id="left">
         <div id="divWordWrap2">
-            <a href="http://www.adm.uwaterloo.ca/infocour/CIR/SA/under.html">{{i.term}} - {{i.course_subject}} {{i.course_id}} Sec {{i.section}} - {{i.course_name}} - {{i.instructor_name}}, {{i.instructor_email}}</a></div></div>
+        <!--This is hard coded for MSCI and term 1179. Term to be made dynamic.-->
```
use {i.term} instead of 1179 and {i.course_subject} instead of MSCI

j2kan commented on taworks/taform/templates/taform/application.html:
```diff
@@ -109,31 +143,35 @@
     For each course that you have held a TA position in the past three terms, please provide the course name, instructor and your responsibilities during that time: 
 </div><br>
-Experience 1:<br>
+TA Positions Held Previous Fall Term:<br>
```
Try <p> </p> tags i think it can replace all the break tags you have

amyleblond replied:
not sure what you mean

j2kan replied:
'<p> </p>'

j2kan replied:
<            p          >              </                    p >

j2kan commented on taworks/taform/templates/taform/application.html:
```diff
@@ -109,31 +143,35 @@
+    In addition to the above information, an application for a teaching assistant position implies that the applicant agrees to the following conditions of employment, and gives the department permission to verify that the student meets these conditions via examination of the student record.
+</div>
+<div id="divWordWrap">
+For full details on the Responsibilities and Requirements of a Management Sciences Teaching Assistantship please read in full <a href=https://uwaterloo.ca/management-sciences/graduate-students/teaching-assistantships-management-sciences>teaching assistantships management sciences</a>.
```
make this open on a new page or else they can't go back again hahaha

j2kan commented on taworks/taform/templates/taform/application.html:
```diff
@@ -109,31 +143,35 @@
-        <li>I confirm that I do not have any outstanding English Language Proficiency requirements as part of my conditions of admission for my Graduate Studies program.</li>
-        <li>I understand that as a Full-Time registered graduate students I am not permitted to be employed by the University for more than an average of (10) ten hours per week per term.</li>
-        <li>I acknowledge that I will be available from the first day of the first month of the term to the last day of the last month of the term.</li>
+        <li><input type="checkbox" id="check1" onchange="document.getElementById('check2').disabled = !this.checked;">&nbsp;I confirm that I do not have any outstanding English Language Proficiency requirements as part of my conditions of admission for my Graduate Studies program.</li><br>
```
Can you explain the logic here with the id="check1" but the document.getElementByID('check2').disabled.

amyleblond replied:
have to check them all before they can check the last for agree and submit

j2kan commented on taworks/taform/views.py:
```diff
@@ -27,7 +27,10 @@ def logout(request):
+    return render(django_logout(request), 'registration/login.html') 
+
+def intro(request): 
+    return render(request, 'taform/intro.html')  
```
need to add logic when user submits form, redirect to apply page

if request.method == 'POST':

amyleblond replied:
don't understand this

j2kan commented on taworks/taform/templates/taform/intro.html:
```diff
@@ -0,0 +1,89 @@
+
+<p>After the application period is closed the department will review all applications and successful candidates will be notified by the department via email. </p>
+
+<input type="submit" name="moveOn" id="moveOn" value="Proceed to Application">
```
this doesn't take you to the next page right now does it?

amyleblond replied:
yesss

lauradobson commented on taworks/taform/templates/taform/application.html:
```diff
@@ -96,7 +129,8 @@
         <div id="divWordWrap2">
-            <a href="http://www.adm.uwaterloo.ca/infocour/CIR/SA/under.html">{{i.term}} - {{i.course_subject}} {{i.course_id}} Sec {{i.section}} - {{i.course_name}} - {{i.instructor_name}}, {{i.instructor_email}}</a></div></div>
+        <!--This is hard coded for MSCI and term 1179. Term to be made dynamic.-->
+            <a target="_blank" href="http://www.adm.uwaterloo.ca/cgi-bin/cgiwrap/infocour/salook.pl?sess=1179&subject=MSCI&cournum={{i.course_id}}">{{i.course_subject}} {{i.course_id}} Sec {{i.section}} - {{i.course_name}} - {{i.instructor_name}}</a></div></div>
```
Glad you got this to work!!

amyleblond committed:
adding changes as per PR

amyleblond committed:
merge from master
Merge branch 'master' of https://github.com/j2kan/ta-works into Frontend_Iter2

amyleblond committed:
final changes for first sprint

amyleblond committed:
fix intro form
final changes for first sprint

amyleblond committed:
go
Merge branch 'Frontend_Iter2' of https://github.com/j2kan/ta-works into Frontend_Iter2

amyleblond committed:
go
Merge branch 'Frontend_Iter2' of https://github.com/j2kan/ta-works into Frontend_Iter2

amyleblond committed:
Merge branch 'Frontend_Iter2' of https://github.com/j2kan/ta-works into Frontend_Iter2

amyleblond committed:
Merge branch 'Frontend_Iter2' of https://github.com/j2kan/ta-works into Frontend_Iter2

amyleblond committed:
Merge branch 'Frontend_Iter2' of https://github.com/j2kan/ta-works into Frontend_Iter2
